mesonbuild/backend/ninjabackend.py

```diff
@@ -1396,69 +1411,56 @@ def generate_swift_target(self, target, outfile):
         # Introspection information
         self.create_target_source_introspection(target, swiftc, compile_args + header_imports + module_includes, relsrc, rel_generated)
 
-    def generate_static_link_rules(self, is_cross, outfile):
+    def generate_static_link_rules(self, outfile):
         num_pools = self.environment.coredata.backend_options['backend_max_links'].value
-        if 'java' in self.build.compilers:
-            if not is_cross:
-                self.generate_java_link(outfile)
-        if is_cross:
-            if self.environment.is_cross_build():
-                static_linker = self.build.static_cross_linker
-            else:
-                static_linker = self.build.static_linker
-            crstr = '_CROSS'
-        else:
-            static_linker = self.build.static_linker
-            crstr = ''
-        if static_linker is None:
-            return
-        rule = 'rule STATIC%s_LINKER\n' % crstr
-        if static_linker.can_linker_accept_rsp():
-            command_template = ''' command = {executable} $LINK_ARGS {output_args} @$out.rsp
+
+        if 'java' in self.build.compilers.host:
+            self.generate_java_link(outfile)
+        for for_machine in list(MachineChoice):
```

Comment:
do you actually need to make this concrete?

Reply:
Do you mean I could use `iter(MachineChoice)` or something?

Reply:
`enum.EnumMeta` implements a `__iter__()` method, so I don't think we need to do anything more than `for for_machine in MachineChoice`

Reply:
Ah right. Forgot about those `__iter__`.